### PR TITLE
UX: Display total fiat balance on home screen

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

Calculates and displays the total fiat value of a given account's native currency and tokens.

* Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1191

## Screenshots/Screencaps

SCR-20230918-nvzn

SCR-20230918-nyzh

## Manual Testing Steps

1.  Look at an account with native currency and token balances
2. See correct balance displayed above token list
3. Switch chains using the network picker
4. Ensure balance on L2 is correct

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by vthomas13 — DISMISSED** [review]

**Review by NidhiKJha — COMMENTED** [review]
Excellent!!
- Checked with native balance
- Checked with detected tokens and overall balance is correct

But we need the loading state here, it should not be removed

**Review by NidhiKJha — DISMISSED** [review]
Awesome!!

**Review by vthomas13 — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]

File: shared/modules/conversion.utils.ts [hunk 1]
```diff
@@ -156,6 +156,16 @@ export function sumHexes(first: string, ...args: string[]) {
   return total.toPrefixedHexString();
 }
 
+export function sumDecimals(first: string, ...args: string[]) {
```
[the remark this answers is not in the document]
Comment (darkwing) [reply shared/modules/conversion.utils.ts:159]: No, I put it here next to `sumHexes` as it essentially does the same thing but for decimals.

File: shared/modules/conversion.utils.ts [hunk 2]
```diff
@@ -156,6 +156,16 @@ export function sumHexes(first: string, ...args: string[]) {
...
+  const firstValue = new Numeric(first, 10);
```
[the remark this answers is not in the document]
Comment (darkwing) [reply shared/modules/conversion.utils.ts:160]: Tests added!

**Review by DDDDDanica — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**